### DISCUSSION TITLE
Shebang requirement and `cargo` macro

### DISCUSSION THREAD

**Discussion by tgross35** [epage/cargo-script-mvs#160] [Ideas]
All the examples have the cargo shebang - is this a requirement to identify a cargo-script file? If so, presence of this shebang (or a fallback option like `--as-cargo-script`) could be used to indicate to `rustc` that it should ignore the special `cargo!` macro or attribute.

This would be an alternative to `std` containing the `cargo` macro, as mentioned here https://github.com/epage/cargo-script-mvs/blob/b31a41c43634acc8d2358811ba4706d1f3dff4fc/0000-cargo-script.md#embedded-manifest-format

Another more general option could be for `rustc` to gain a `--ignore-macros` option that, well, ignores macros. This would mean that it doesn't need any specific awareness of cargo-script. Additionally, it would open up other tools to doing in-source configuration in a similar way: which could be either good or bad.

**Comment by epage** [comment]
At the moment, shebang is not required.  It might be useful for tools like rust-analyzer to pick up the role of these files.  I have not looked too deeply into shebang detection but my gut says its a messy afair based on [perl](https://stackoverflow.com/questions/38059830/how-does-perl-avoid-shebang-loops) so I'd prefer for us to not make it a requirement.  I also am considering `cargo Cargo.toml` support for complete consistency in where manifests are supported and that means the shebang isn't a complete tell.

**Reply by tgross35** [reply]
Fair enough, the idea just came up because I believe the parser has to extract the shebang separately (at least `syn` does) so it could provide a hint - but not relying on it makes more sense.
